libcxx/test/std/experimental/simd/simd.class/simd_unary.pass.cpp

```diff
@@ -0,0 +1,159 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// UNSUPPORTED: c++03, c++11, c++14
+
+// <experimental/simd>
+//
+// [simd.class]
+// simd& operator++() noexcept;
+// simd operator++(int) noexcept;
+// simd& operator--() noexcept;
+// simd operator--(int) noexcept;
+// mask_type operator!() const noexcept;
+// simd operator~() const noexcept;
+// simd operator+() const noexcept;
+// simd operator-() const noexcept;
+
+#include "../test_utils.h"
+#include <experimental/simd>
+
+namespace ex = std::experimental::parallelism_v2;
+struct CheckSimdUnaryOperator {
+  template <class _Tp, class SimdAbi, std::size_t _Np>
+  void operator()() {
+    constexpr size_t array_size = ex::simd_size_v<_Tp, SimdAbi>;
+    {
+      ex::simd<_Tp, SimdAbi> origin_simd([](_Tp i) { return i; });
+
+      std::array<_Tp, array_size> origin_value;
+      for (size_t i = 0; i < array_size; i++)
+        origin_value[i] = static_cast<_Tp>(i);
+
+      auto after_increment_simd = ++origin_simd;
+
```

Comment:
Missing type check for `after_increment_simd`. Please check all tests for such problems. I will not mark them again